src/Data/TypeRep/TMap.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+
+-- | This module introduces 'TMap' data type — 'TR.TypeRepMap' parametrized by 'Identity'.
+
+module Data.TypeRep.TMap
+       ( -- * Map type
+         TMap
+
+         -- * Construction
+       , TR.empty
+       , one
+
+         -- * Modification
+       , insert
+       , delete
+       , TR.hoist
+       , TR.unionWith
+       , TR.union
+
+         -- * Query
+       , lookup
+       , member
+       , TR.size
```

Comment:
Maybe it's better to have all types be `TMap` in documentation? To prevent confusion. I like idea of not introducing too much boilerplate. But having good documentation is also important.